### PR TITLE
Update query documentation with new array query features

### PR DESCRIPTION
**Pull request by arthurprs**
From https://github.com/getditto/ditto/pull/4317

### CODE REVIEW

File: docs/concepts/querying.md [hunk 1]
```diff
@@ -328,6 +341,35 @@ Use `sum(propertyPath)` to return a summation float that represents the sum of t
...
+```
+
+Use `array[*].subpath` that collects the subpath of each array item in the result array.
+
+Eg. `books[*].authors[0]` for the example document is `["J. K. Rowling", "J. R. R. Tolkien", "George R. R. Martin", "Stephen King"]`
+
+Eg. `books[*].authors` for the example document is `[["J. K. Rowling"], ["J. R. R. Tolkien"], ["George R. R. Martin"], ["Stephen King", "Peter Straub"]]`
+  
+```js
+// finds documents with books whose Stephen King is the first author
```
Comment (russelldb) [inline docs/concepts/querying.md:360]:
```suggestion
// finds documents with books whose first author is Stephen King
```

File: docs/concepts/querying.md [hunk 2]
```diff
@@ -328,6 +341,35 @@ Use `sum(propertyPath)` to return a summation float that represents the sum of t
...
+Use `array[? expression]` to return an array containing only the input array items that satisfy the expression.
+  
+Eg. `books[? length(authors) >= 2]` for the example document is `[{"title": "The Talisman", "authors": ["Stephen King", "Peter Straub"]}]`.
+
+```js
+// finds documents with books that have 2 or more authors
+find("books[? length(authors) >= 2] != []")
+```
+
+Use `array[*].subpath` that collects the subpath of each array item in the result array.
```
Comment (russelldb) [inline docs/concepts/querying.md:353]:
```suggestion
Use `array[*].subpath` to collects the subpath of each array item in the result array.
```

**Review by russelldb — APPROVED** [review]
Left a couple of suggestions, which I hope are improvements (it is Friday afternoon, I may have misread!)

File: docs/concepts/querying.md (hunk 1 above)
Comment (arthurprs) [reply docs/concepts/querying.md]: Thanks!

**Review by russelldb — APPROVED** [review]
👍

**Review by hamchapman — APPROVED** [review]

**Pull request merged**